Documentation: add personal access token documentation (#4258)
mozilla/pontoon: documentation/docs/localizer/users.md

![LOCALES tab in translation tools](../assets/localizer/users/translation_locales.png "LOCALES tab in translation tools")

> ### Personal Access Tokens

Reviewer: ```suggestion
### Personal Access Tokens

```

> Users can create Personal Access Tokens (PAT) which are required for accessing all endpoints of the  Pontoon [REST API](https://github.com/mozilla/pontoon/blob/main/pontoon/api/README.md).

Reviewer: ```suggestion
Users can create Personal Access Tokens (PAT) which are required for accessing all endpoints of the Pontoon [REST API](https://github.com/mozilla/pontoon/blob/main/pontoon/api/README.md).
```

Not all of them?

Reviewer 2: Just pretranslation and user actions.

> To create a PAT, the user must enter a suitable token name no longer than 32 characters. The generated token is used with the REST API as follows:

Reviewer: ```suggestion
To create a PAT, the user must enter a suitable token name no longer than 32 characters. The generated token can be used with the REST API as follows:
```
